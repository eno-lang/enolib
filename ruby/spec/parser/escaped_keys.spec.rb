# frozen_string_literal: true

input = <<~DOC.strip
`key`:
``k`ey``:
```ke``y```    :
    `` `key` ``    :
DOC

describe Enolib::Parser do
  describe 'Escaped keys' do
    context = Enolib::Context.new(input)

    it 'parses as expected' do
      expect(context.document).to match_snapshot
    end
  end
end
