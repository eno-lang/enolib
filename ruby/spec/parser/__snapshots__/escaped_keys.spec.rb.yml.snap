---
"./spec/parser/escaped_keys.spec.rb[1:1:1]": &1
  :elements:
  - :index: 0
    :line: 0
    :ranges:
      :line:
      - 0
      - 6
      :field_operator:
      - 5
      - 6
      :escape_begin_operator:
      - 0
      - 1
      :escape_end_operator:
      - 4
      - 5
      :key:
      - 1
      - 4
    :key: key
    :type: :field_or_fieldset_or_list
    :parent: *1
  - :index: 7
    :line: 1
    :ranges:
      :line:
      - 7
      - 16
      :field_operator:
      - 15
      - 16
      :escape_begin_operator:
      - 7
      - 9
      :escape_end_operator:
      - 13
      - 15
      :key:
      - 9
      - 13
    :key: k`ey
    :type: :field_or_fieldset_or_list
    :parent: *1
  - :index: 17
    :line: 2
    :ranges:
      :line:
      - 17
      - 33
      :field_operator:
      - 32
      - 33
      :escape_begin_operator:
      - 17
      - 20
      :escape_end_operator:
      - 25
      - 28
      :key:
      - 20
      - 25
    :key: ke``y
    :type: :field_or_fieldset_or_list
    :parent: *1
  - :index: 34
    :line: 3
    :ranges:
      :line:
      - 34
      - 54
      :field_operator:
      - 53
      - 54
      :escape_begin_operator:
      - 38
      - 40
      :escape_end_operator:
      - 47
      - 49
      :key:
      - 41
      - 46
    :key: "`key`"
    :type: :field_or_fieldset_or_list
    :parent: *1
  :type: :document
